### ISSUE TITLE
Data failed to pass the specified data quality criteria

### ISSUE THREAD

**Issue by GatutDaniarsyad** [QuakeMigrate/QuakeMigrate#167]
**Describe the bug**
Hi, I'm facing problem when running the detection of QM code. It says: "All data for this timestep failed to pass the specified data quality criteria. This includes the presence of gaps or overlaps, or the data not spanning the full time window". I have check the data and remove all with gappy data. I also tried using shorter duration from 1 day to 10 minutes of data but it didn't resolve the problem. I have set the _allow _gaps__ parameter to True and resulted in the same error.

**To Reproduce**
Here is my code:
```
# --- Set time period over which to run detect ---
starttime = "2023-12-13T23:30:00.0"
endtime = "2023-12-13T23:40:00.0"`

# --- Read in station file ---
stations = read_stations(station_file)

# --- Create new Archive and set path structure ---
archive = Archive(archive_path=data_in, stations=stations,
                  archive_format="YEAR/JD/STATION")

# --- Load the LUT ---
lut = read_lut(lut_file=lut_file)
lut.decimate([2, 2, 2], inplace=True)

# --- Create new Onset ---
onset = STALTAOnset(position="classic", sampling_rate=20)
onset.phases = ["P", "S"]
onset.bandpass_filters = {
    "P": [1, 8, 2],
    "S": [1, 8, 2]}
onset.sta_lta_windows = {
    "P": [0.4, 2.0],
    "S": [1.0, 2.0]}
onset.allow_gaps = True

# --- Create new QuakeScan ---
scan = QuakeScan(archive, lut, onset=onset, run_path=run_path,
                 run_name=run_name, log=True, loglevel="info")

# --- Set detect parameters ---
scan.timestep = 300.
scan.threads = 4  # NOTE: increase as your system allows to increase speed!

# --- Run detect ---
scan.detect(starttime, endtime)
```

And here is the error:
```
==============================================================================================================
        DETECT - Continuous coalescence scan
==============================================================================================================

        Scanning from 2023-12-13T23:30:00.000000Z to 2023-12-13T23:40:00.000000Z

        Scan parameters:
                Scan sampling rate = 20 Hz
                Thread count       = 4
                Time step          = 300.0 s

        Onset parameters - using the classic STA/LTA onset
                Onset function sampling rate = 20 Hz
                Phase(s) = ['P', 'S']

                P bandpass filter  = [1, 8, 2] (Hz, Hz, -)
                S bandpass filter  = [1, 8, 2] (Hz, Hz, -)

                P onset [STA, LTA] = [0.4, 2.0] (s, s)
                S onset [STA, LTA] = [1.0, 2.0] (s, s)

==============================================================================================================
~~~~~~~~~~~~~~~~~~~~ Processing : 2023-12-13T23:30:00.000000Z-2023-12-13T23:35:00.000000Z ~~~~~~~~~~~~~~~~~~~~
                No P onset for IA.CBJI.
                No P onset for IA.JTJM.
                No P onset for IA.PSJM.
                No P onset for IA.SADLY.
                No P onset for IA.SSJM.
                No P onset for IA.WSJM.
                No S onset for IA.CBJI.
                No S onset for IA.JTJM.
                No S onset for IA.PSJM.
                No S onset for IA.SADLY.
                No S onset for IA.SSJM.
                No S onset for IA.WSJM.
                All data for this timestep failed to pass the
                specified data quality criteria. This includes the
                presence of gaps or overlaps, or the data not
                spanning the full time window.
~~~~~~~~~~~~~~~~~~~~ Processing : 2023-12-13T23:35:00.000000Z-2023-12-13T23:40:00.000000Z ~~~~~~~~~~~~~~~~~~~~
                No P onset for IA.CBJI.
                No P onset for IA.JTJM.
                No P onset for IA.PSJM.
                No P onset for IA.SADLY.
                No P onset for IA.SSJM.
                No P onset for IA.WSJM.
                No S onset for IA.CBJI.
                No S onset for IA.JTJM.
                No S onset for IA.PSJM.
                No S onset for IA.SADLY.
                No S onset for IA.SSJM.
                No S onset for IA.WSJM.
                All data for this timestep failed to pass the
                specified data quality criteria. This includes the
                presence of gaps or overlaps, or the data not
                spanning the full time window.
==============================================================================================================
```

**Expected behavior**
I expect to can detect the event using this data, or if there any pre-processing which can overcome this problem so that the data can be used properly.

**Desktop (please complete the following information):**
 - Operating System: WSL Ubuntu-20.04 / Windows 11
 - Python version: 3.9.18
 - QuakeMigrate version: 1.0.2

**Comment by hemmelig** [comment]
Hi—thanks for getting in touch! My quick take is that this is likely due to the format of the archive your data are stored in.

What is the directory structure/file name format you are using for your stored data? Right now, you are using one of the built-in options `YEAR/JD/STATION`—this means it expects to see something like the following underneath the `data_in` path you have specified:

```
2023/347/JTJM.m
```

If you files are not stored in this format, you can instead do something like:

```python
# --- Create new Archive and set path structure ---
archive = Archive(
    archive_path=data_in,
    stations=stations,
)
archive.format = "{year}/{jday:03d}/{station}"
```

where you need to specify a Python f-string. You can use any of the following keys: `year`, `month`, `day`, `jday`, `station`, and `dtime`. I can help create this for you if needed!

If this isn't the issue, we can continue to diagnose from there.

**Comment by GatutDaniarsyad** [comment]
Hi! I have successfully run the detect.py.

I found that the error is because I set the time period from the start of the data. My data is *.2023-12-13T23:30:00.000000Z-2023-12-13T23:40:00.000000Z. It will result in an error if the time period is from the start (e.g starttime = "2023-12-13T23:30:00.0") or until the end of the data.

If the code cannot find the data, it will say: "No files found in archive for this time period."

Thanks for the quick response!
